tests/test_view_functions.py

```diff
@@ -60,6 +63,71 @@ def test_event_view_func_wrapper(monkeypatch):
     ]
 
 
+def test_run_cloudevent():
+    headers = {"Content-Type": "application/cloudevents+json"}
+    data = json.dumps(
+        {
+            "source": "from-galaxy-far-far-away",
+            "type": "cloudevent.greet.you",
+            "specversion": "1.0",
+            "id": "f6a65fcd-eed2-429d-9f71-ec0663d83025",
+            "time": "2020-08-13T02:12:14.946587+00:00",
+            "data": {"name": "john"},
+        }
+    )
+    request = pretend.stub(headers=headers, get_data=lambda: data)
+    function = pretend.call_recorder(lambda cloudevent: "hello")
+    functions_framework._run_cloudevent(function, request)
+    event = function.calls[0].__dict__["args"][0]
+    assert event.data == {"name": "john"}
+    assert event["id"] == "f6a65fcd-eed2-429d-9f71-ec0663d83025"
```

Comment:
This feels pretty brittle to me. I would expect something like:

```python
expected_cloudevent = CloudEvent(data=data["data"]..., id=data["id"]) # or whatever
assert function.calls == [pretend.call(expected_cloudevent)]
```
This ensures that the function is only being called once, that we're reusing the same data from the setup portion of the test (rather than duplicating it), etc.

Reply:
Yeah,  we originally launched version [1.0.1](https://pypi.org/project/cloudevents/1.0.1/) (currently yanked) to implement an equality override so we can add the code you wrote above here :D